### DISCUSSION TITLE
Please explain what this project is capable or/built for in easiest way possible with real life examples

### DISCUSSION THREAD

**Discussion by Greatz08** [dora-rs/dora#466] [General]
To be honest I couldn't understand much from main page of project as it directly starts from
Why Dora?
But I believe you should be explaining what is dora-rs in easiest and detailed way and then Why Dora part would make sense plus I wanted to see like what can be achieved with it. As it is framework many things I believe might be possible so in real life cases what all things can be created or what problema can be solved with this and maybe then some detailed EASY tutorialsso that max can understand and relate and understand true potential of this project.I asked for this because I haven't personal used ros or similar thing so I don't know anything about this project and many will be in same spot so please explain each and every thing in easiest way in detail with example so that max can understand and contribute too if possible

**Comment by haixuanTao** [comment]
That makes sense. I'm redoing the `README` in #463 and i'll make sure to include a part on practicallly what can dora-rs do.

In all honesty, dora-rs is still experimental and you might expect a bit of bug. We're trying our hardest to fix all issues but might take a bit of time.

To answer you're question: dora-rs makes DIY robotics possible. The reason why you can't just use a python script is that you might have multiple sensors and computation running in parallel in different languages and you need a way to compose them together. Now Python is awful when it comes to parallelism as the GIL, and thread-safety is not guaranteed. You can use Rust or C/C++ but then doing Deep Learning is super hard. And then you might want to have cloud services on top. 

We tries to solve all those problems by giving you an abstraction layer as a dataflow , and will take care of passing messages along the dataflow regardless of the languages of each nodes and if its distributed or not. All that with minimal latency. 

Then we have additional services like Opentelemetry, or the fact that you can record all your data as parquet file that you can later reuse for training/finetuning pushed to huggingface ...

Or the fact that you can develop your app using hot-reloading combined with an LLM to let him be programmed in natural languages.

You can check out #463 and leave a review on it if parts are not super clear :) 

Genuinely appreciate that you left a discussions here and hope that i answered your question.

**Comment by haixuanTao** [comment]
We merged #463 . Hopefully what is dora-rs should make more sense now.
